packages/rest-typings/src/v1/omnichannel.ts

```diff
@@ -224,4 +233,325 @@ export type OmnichannelEndpoints = {
+			message: {
+				_id: string;
+				ts: string;
+			};
+		};
+	};
+
+	'livechat/messages.history/:rid': {
+		GET: (params: { rid: string; searchText: { text: string }; token: string; ls: string; end: string; limit: string }) => {
+			messages: IMessage[];
+		};
+	};
+
+	'livechat/messages': {
+		POST: (params: { visitor: { token: string }; messages: { msg: string }[] }) => {
+			messages: { username: string; msg: string; ts: Date }[];
+		};
+	};
+
+	'livechat/offline.message': {
+		POST: (params: { name: string; email: string; message: string; department?: string; host?: string }) => { message: string };
+	};
+
+	'livechat/room.close': {
+		POST: (params: { rid: string; token: string }) => { rid: string; comment: string };
+	};
+
+	'livechat/room.transfer': {
+		POST: (params: { rid: string; token: string; department: string }) => { room: IOmnichannelRoom };
+	};
+
+	'livechat/room.visitor': {
+		PUT: (params: { rid: string; oldVisitorId: string; newVisitorId: string }) => { room: IOmnichannelRoom };
+	};
+
+	'livechat/transcript': {
+		POST: (params: { token: string; rid: string; email: string; user: any; subject: string }) => { message: string };
+	};
+
+	'livechat/transfer.history/:rid': {
+		GET: (params: { rid: string }) => { history: number[]; count: number; offset: number; total: number };
+	};
+
+	'livechat/video.call/:token': {
+		GET: (params: { token: string; rid?: string }) => {
+			videoCall: {
+				rid: string;
+				domain: SettingValue;
+				provider: string;
+				room: string;
+				timeout: Date;
+			};
+		};
+	};
+
+	'livechat/webrtc.call': {
+		GET: (params: { rid?: string }) => { videoCall: { rid: any; provider: string; callStatus: any } };
+	};
+
+	'livechat/webrtc.call/:callId': {
+		PUT: (params: { callId: string; rid?: string; status?: string }) => { status: string | undefined };
+	};
+
+	'livechat/facebook': {
+		POST: (params: {
+			mid: string;
+			page: string;
+			token: string;
+			first_name: string;
+			last_name: string;
+			text: string;
+			attachments: any[];
+		}) => { success: boolean; error?: string; message?: any };
+	};
+
+	'livechat/visitors.pagesVisited/:roomId': {
+		GET: (params: PaginatedRequest<{ roomId?: string }>) => PaginatedResult<{ pages: IMessage[] }>;
+	};
+
+	'livechat/visitors.chatHistory/room/:roomId/visitor/:visitorId': {
+		GET: (params: PaginatedRequest<{ roomId?: string; visitorId?: string }>) => PaginatedResult<{ history: IMessage[] }>;
+	};
+
+	'livechat/visitors.searchChats/room/:roomId/visitor/:visitorId': {
+		GET: (
+			params: PaginatedRequest<{
+				roomId?: string;
+				visitorId?: string;
+				searchText?: string;
+				closedChatsOnly?: booleanString;
+				servedChatsOnly?: booleanString;
+			}>,
+		) => PaginatedResult<{ history: IMessage[] }>;
+	};
+
+	'livechat/visitors.autocomplete': {
+		// selector object is of type => { exceptions?: string[]; conditions?: FilterQuery<ILivechatVisitor>; term: string } }
+		GET: (params: PaginatedRequest<{ selector: string }>) => {
```

Comment:
There's a comment saying that selector is { exceptions: [], conditions: [] }
why you typed it as string? 🙈 
Also, `term` seems to be missing from params

Reply:
:see_no_evil:  Because it is indeed a string of the json stringified object. The term param is also encoded within the same json object. TBH, I didn't like this approach of strigifying the object in query params, it could have been separate individual params....

Reply:
It seems this endpoint is only getting used in one place throughout the code base & it is only passing the `term` param - so the other 2 params (i.e. exceptions, conditions) is aren't getting used. So I think it should be safe to deprecate this endpoint in future versions & further optimize the search query.

Reply:
https://github.com/RocketChat/Rocket.Chat/pull/25344#issuecomment-1119509496